src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -528,6 +530,24 @@ private Task ProcessCompilationUnitCompleted(CompilationUnitCompletedEvent compl
             }
         }
 
+        private async Task ExecuteCompilationActions(ImmutableDictionary<DiagnosticAnalyzer, ImmutableArray<CompilationAnalyzerAction>> compilationActionsMap, CancellationToken cancellationToken)
+        {
+            // Execute analyzers in parallel.
+            var tasks = ArrayBuilder<Task>.GetInstance();
+            foreach (var analyzerAndActions in compilationActionsMap)
+            {
+                var task = Task.Run(() =>
```

Comment:
This method is in the compiler AnalyzerDriver, the ExecuteCompilationEndActions on AnalyzerExecutor (which is shared between drivers) always runs actions sequentially.